integration-tests/testdb.go

```
@@ -0,0 +1,148 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to create database %s: %w", dbName, err)
+	}
+
+	testConnStr := dbConfig.ToConnectionStr()
+	testDb, err := sql.Open("postgres", testConnStr)
+	if err != nil {
+		adminDb.Exec(fmt.Sprintf(`DROP DATABASE "%s" WITH (FORCE)`, dbName))
+		return nil, fmt.Errorf("failed to connect to test database: %w", err)
+	}
+
+	if err := testDb.Ping(); err != nil {
```

golangci-lint (2.11.4)

[error] (*database/sql.DB).Ping must not be called. use (*database/sql.DB).PingContext

(noctx)